modules/tasks/tasks/src/test/java/org/motechproject/tasks/service/impl/TaskServiceImplTest.java

```diff
@@ -192,6 +193,16 @@ public void shouldNotSaveTaskIfNotMatchExtraValidationConditions() {
         taskService.save(task);
     }
 
+    @Test(expected = TaskNameAlreadyExistsException.class)
+    public void shouldNotSaveTaskWithDuplicateName() {
+
+        when(tasksDataService.findTasksByName("name")).thenReturn(
+                asList(new Task("name", trigger, asList(action))));
+
+        Task t = new Task("name", trigger, asList(action), null, false, false);
+        taskService.save(t);
```

Comment:
Can we also verify in the finally block, that there were no more interactions with the TasksDataService?